### PR TITLE
Support SharedLayer with different prefixs for pipeline parallel

### PR DESCRIPTION
**Pull request by DrownFish19**
### PR types

New features

### PR changes

Others

### Description

Support SharedLayer with different prefixs.

Currently, SharedLayers only adopt the prefix of the first layer. However, the different prefixes are required by different model structures. This PR supports the definition of different prefixes for sharedlayers.

### CODE REVIEW

File: paddlenlp/transformers/gpt/modeling.py [hunk 1]
```diff
@@ -1198,7 +1198,7 @@ class GPTPretrainingCriterion(paddle.nn.Layer):
     def __init__(self, config):
         super(GPTPretrainingCriterion, self).__init__()
         self.config = config
-        if config.tensor_parallel_degree > 1 and config.tensor_parallel_output:
+        if config.tensor_parallel_degree > 1 and config.tensor_parallel_output and not config.sequence_parallel:
```
Comment (DrownFish19) [inline paddlenlp/transformers/gpt/modeling.py]: 因为使用sequence_parallel之后，在LMHead已经通过GatherOP聚合参数，所以输出的logits应该是聚合后的，此处不应使用ParallelCrossEntropy。

Comment (ZHUI) [reply paddlenlp/transformers/gpt/modeling.py]: 不太对吧。这样会损失计算效率的。

File: paddlenlp/transformers/model_utils.py [hunk 2]
```diff
@@ -2420,43 +2421,35 @@ def _set_pipeline_name_mapping(self, mappings=None):
...
+                            single_name = [prefixes[idx]]
                             single_name.extend(name_splited[2:])
                         else:
-                            single_name = [prefixs[str(len(prefixs) - 1)]]
+                            single_name = [prefixes[str(len(prefixes) - 1)]]
                             single_name.extend(name_splited[2:])
                             logger.warning(
                                 f"Please check! we treat this key as last layer, get {k}, set origin name as {'.'.join(single_name)}"
                             )
                     elif name_splited[0] == "shared_layers":
```
Comment (ZHUI) [inline paddlenlp/transformers/model_utils.py:2438]: get_sequential_name_prefixs 这个不需要适配 shared_layers 的修改吗？

Comment (DrownFish19) [reply paddlenlp/transformers/model_utils.py:2438]: get_sequential_name_prefixs会拿到所有层的prefixs,之前的结果是正确的，不需要进行修改

File: paddlenlp/transformers/gpt/modeling.py (hunk 1 above)
Comment (DrownFish19) [reply paddlenlp/transformers/gpt/modeling.py]: 这部分逻辑之前理解错误，已经修改

File: paddlenlp/transformers/model_utils.py (hunk 2 above)
Comment (DrownFish19) [reply paddlenlp/transformers/model_utils.py:2438]: 后续修改get_sequential_name_prefixs为get_sequential_name_prefixes，仅名称修改

**Review by ZHUI — DISMISSED** [review]
LGTM

**Review by ZHUI — APPROVED** [review]
LGTM

**Pull request merged**